src/net/sourceforge/kolmafia/objectpool/Concoction.java

```diff
@@ -414,6 +414,10 @@ public int compareTo(final Concoction o) {
       throw new NullPointerException();
     }
 
+    if (this == o) {
```

Comment:
Somewhere I picked up the idea that == in compareTo was not a best practice so I avoided this.  I don't think this is what we want and will try and get a test.  I think if I clone a steel item and then compare the original to the clone this compareTo will not return 0 and that is wrong.  Stand by, please.

Reply:
Do we not just want to use `.equals()` then instead? Seems like the right move